### PR TITLE
programs.gnome-terminal: terminal options

### PR DESCRIPTION
**Pull request by kamadorueda**
- Add support for showing bold as bright colors
- Add support to configure the background transparency
- Fix the scrollOnOutput, it was not being dumped to the config
- Add tests!
- Add myself as maintainer
- image
- demo

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/gnome-terminal.nix [hunk 1]
```diff
@@ -234,13 +240,20 @@ let
         type = types.bool;
         description = "Turn on/off the terminal's bell.";
       };
+
+      transparencyPercent = mkOption {
+        default = null;
+        type = types.nullOr types.int;
+        description = "Background transparency percentage (0-100).";
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix:247]: Can use a more precise type here
```suggestion
        type = types.nullOr (types.between 0 100);
        description = "Background transparency in percent.";
```

**Comment by rycee** [comment]
Thanks for the contribution! I added a minor comment.

File: tests/modules/programs/gnome-terminal/gnome-terminal-1.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          scrollbackLines = 1000000;
+          scrollOnOutput = false;
+          showScrollbar = false;
+          transparencyPercent = 5;
+          visibleName = "kamadorueda";
+        };
+      };
+      showMenubar = false;
+    };
+
```
Comment (rycee) [inline tests/modules/programs/gnome-terminal/gnome-terminal-1.nix:50]: Oh, I think there will have to be an overlay here that replaces `gnome3.gnome-terminal` by a dummy package.

Comment (kamadorueda) [reply tests/modules/programs/gnome-terminal/gnome-terminal-1.nix:50]: May I know why? is it for performance reasons?

File: modules/programs/gnome-terminal.nix (hunk 1 above)
Comment (kamadorueda) [reply modules/programs/gnome-terminal.nix:247]: This type may not exist in many nixpkgs version:

image

Comment (rycee) [reply modules/programs/gnome-terminal.nix]: Try `types.ints.between`.

File: tests/modules/programs/gnome-terminal/gnome-terminal-1.nix (hunk 2 above)
Comment (rycee) [reply tests/modules/programs/gnome-terminal/gnome-terminal-1.nix:50]: Yes, it is unnecessary to download the entire Gnome Terminal closure when its not actually used. You can see examples in the other tests by searching for "dummy".

**Comment by kamadorueda** [comment]
I think this PR is now good quality

**Review by berbiche — APPROVED** [review]
LGTM!

**Comment by kamadorueda** [comment]
can we merge guys?

**Comment by berbiche** [comment]
> can we merge guys?

This needs a rebase to fix the CI

**Comment by kamadorueda** [comment]
> > can we merge guys?
> 
> This needs a rebase to fix the CI

done

**Comment by kamadorueda** [comment]
Jajaja :(

Is this something I can fix?

```
macos tests:

error: Package ‘gnome-terminal-3.40.1’ in /nix/store/l4s3fi56740rmzwrs3vndwn8p72xhw8f-nixexprs.tar.xz/pkgs/desktops/gnome/core/gnome-terminal/default.nix:45 is not supported on ‘x86_64-darwin’, refusing to evaluate.
```

**Comment by berbiche** [comment]
> Jajaja :(
> 
> Is this something I can fix?
> 
> ```
> macos tests:
> 
> error: Package ‘gnome-terminal-3.40.1’ in /nix/store/l4s3fi56740rmzwrs3vndwn8p72xhw8f-nixexprs.tar.xz/pkgs/desktops/gnome/core/gnome-terminal/default.nix:45 is not supported on ‘x86_64-darwin’, refusing to evaluate.
> ```
> 

The module needs to be gated.
In tests/default.nix, you need to move the test in the section for the tests reserved to Linux.

**Comment by kamadorueda** [comment]
> The module needs to be gated.
> In tests/default.nix, you need to move the test in the section for the tests reserved to Linux.

Thanks! done

**Comment by kamadorueda** [comment]
Can we merge guys? 😝 
Sorry for being so insistent

File: tests/modules/programs/gnome-terminal/gnome-terminal-1.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  config = {
+    nixpkgs.overlays = [
+      (self: super: {
+        gnome3.gnome-terminal =
+          pkgs.writeScriptBin "dummy-gnome3-gnome-terminal" "";
```
Comment (berbiche) [inline tests/modules/programs/gnome-terminal/gnome-terminal-1.nix:10]:
```suggestion
        gnome.gnome-terminal =
          pkgs.writeScriptBin "dummy-gnome3-gnome-terminal" "";
```

Sorry, last minute request.
I was looking at the logs to verify whether additional dependencies were being fetched during the test run.
`gnome3` has been renamed to `gnome`.

**Review by berbiche — CHANGES_REQUESTED** [review]

File: tests/modules/programs/gnome-terminal/gnome-terminal-1.nix (hunk 3 above)
Comment (kamadorueda) [reply tests/modules/programs/gnome-terminal/gnome-terminal-1.nix]: done

**Review by berbiche — APPROVED** [review]
Thanks again.

I'll leave Rycee to comment on the test.

**Comment by rycee** [comment]
Looks good. The test is a bit unorthodox but I think that's OK. I'm working a bit on nmt now and then to make it a bit more convenient for these kinds of cases so maybe it can be cleaned up in the future.

**Comment by kamadorueda** [comment]
> Looks good. The test is a bit unorthodox but I think that's OK. I'm working a bit on nmt now and then to make it a bit more convenient for these kinds of cases so maybe it can be cleaned up in the future.

Sure, let me know how I can help and I'll do more PRs to improve it!

**Pull request merged**